express.js

```diff
@@ -7,9 +7,98 @@ var expressHandlebars = require('express-handlebars');
 var handlebars = require('./handlebars');
 var extendHelpers = require('./src/extend-helpers');
 
+var fs = require('fs');
+var Path = require('path');
+
+var readdirAsync = function(path) {
+	return new Promise(function(res, rej) {
+		fs.readdir(path, function(err, files) {
+			if(err) return rej(err);
+
+			res(files);
+		});
+	});
+};
```

Comment:
You might like denodeify

https://github.com/matthew-andrews/denodeify

#shamelessselfpromotion